polygeist/lib/polygeist/Passes/AffineReduction.cpp

```diff
@@ -291,9 +291,12 @@ class AffineForReductionIter : public OpRewritePattern<AffineForOp> {
 
     // Prepare for new yielded value for 'replaceOp'.
     SmallVector<Value, 4> NewYieldedRes, NewRes(NewForOp.getResults());
-    size_t AdditionalRes =
-        NewForOp.getResults().size() - ForOp.getResults().size();
-    assert(AdditionalRes >= 0 && "must be >= 0");
+    const auto NewForOpNumRes = NewForOp.getNumResults();
```

Comment:
I think that if `NewForOpNumRes` and `ForOpNumRes` are declared with concrete type `size_t` the static_cast at line 299 can be dropped. 

Propose: 
```
size_t NewForOpNumRes = NewForOp.getNumResults();
size_t ForOpNumRes = ForOp.getNumResults();
size_t AdditionalRes = NewForOpNumRes - ForOpNumRes;
```

Reply:
That'd change the types of the first two arguments. I added explicit typing to all of the variables now.